test/error-messages/nested-invalid-value.txt

```diff
@@ -0,0 +1,9 @@
+Invalid value
+
+concat :: Array a -> Array a -> Array a
+                     ^^^^^^^
+                        1
+
+1)  "ab" :: String
+
+The value at position 1 is not a member of (Array Number).
```

Comment:
Ok I think this did turn out to be simpler than I anticipated (you already had a function dealing with this output, and it was only being called once).

One thing, the `???` seemed a bit conspicuous to me. Kinda the whole point of this was that it doesn't matter and the `???` draws my attention. Instead, we just always return the type variable, so the above error message becomes
```
Invalid value

concat :: Array a -> Array a -> Array a
                     ^^^^^^^
                        1

1)  "ab" :: String

The value at position 1 is not a member of (Array a).
```
and another example from the original test suite
```
Invalid value

unnest :: Array (Array a) -> Array a
          ^^^^^^^^^^^^^^^
                 1

1)  [1, 2, 3] :: (Array Number)

The value at position 1 is not a member of (Array (Array a)).
```